Dockerfile.cherrypick-test

```diff
@@ -0,0 +1,89 @@
+FROM public.ecr.aws/amazonlinux/amazonlinux:latest as builder
```

Comment:
Whys this file added?

Reply:
It takes about an hour to run the full `make release` and if there's something wrong with the cherrypicks like a merge conflict it takes multiple hours to resolve.

This file intends to create a quick way to test the cherrypicks compile, like ~5-10 mins.

Reply:
I see. That's cool but let's move this work to a different PR? 

Also, it doesn't take make release an hour to run for me ever. I do it on a C5.something-x-large EC2 instance though.